### ISSUE TITLE
memory leak if not define  TURN_NO_TLS

### ISSUE THREAD

**Issue by huhaipeng** [coturn/coturn#950]
if not define TURN_NO_TLS in file Configure,  then coturn will supoort tls. In this case, if function "register_callback_on_ioa_socket" return -1, function "client_to_be_allocated_timeout_handler" will never be register, so the session will nerver be delete in the session map.

**Comment by huhaipeng** [comment]
Actually, I think if "register_callback_on_ioa_socket" return -1,  the session will never be deleted, so the memory will leak

**Comment by huhaipeng** [comment]
@ggarber

**Comment by huhaipeng** [comment]
![image](https://user-images.githubusercontent.com/23566147/185833125-82dfd90a-80ad-4e8d-98b1-b5cf619237d4.png)

**Comment by huhaipeng** [comment]
![image](https://user-images.githubusercontent.com/23566147/185833307-52581b73-380e-4ebe-bfed-0c3fb3d77d45.png)

**Comment by huhaipeng** [comment]
@ggarber @eakraly

**Comment by eakraly** [comment]
Hi @huhaipeng ! Thank you for the report!
Do you think this is something you can submit a PR for?

**Comment by huhaipeng** [comment]
> Hi @huhaipeng ! Thank you for the report! Do you think this is something you can submit a PR for?

Yes, I have submit a PR for fixing this bug. the commit as follow:
[fix memory leak of turn session]( https://github.com/coturn/coturn/pull/956)
@eakraly

**Comment by huhaipeng** [comment]
I have submit an another  PR for fixing this bug. As follow:
fix turn session leak #962
@eakraly @ggarber

**Comment by ggarber** [comment]
Thank you very much! Merged now.
